src/Raven.Server/Documents/Document.cs

```diff
@@ -102,37 +104,55 @@ public bool Expired(DateTime currentDate)
             return false;
         }
 
-        private static bool IsMetadataEqualTo(BlittableJsonReaderObject currentDocument, BlittableJsonReaderObject targetDocument)
+        private static DocumentCompareResult IsMetadataEqualTo(BlittableJsonReaderObject currentDocument, BlittableJsonReaderObject targetDocument, bool skipAttachment)
```

Comment:
It should be probably moved to BlittableOperation. We have blittable comparison methods there.